Lib/os.py

```diff
@@ -92,6 +94,97 @@ def _get_exports_list(module):
 from os.path import (curdir, pardir, sep, pathsep, defpath, extsep, altsep,
     devnull)
 
+del _names
+
+
+if _exists("_have_functions"):
```

Comment:
We don't need this if block, we implement the `supports_foo` sets in the `posix`/`nt` module.

Reply:
[I wonder what it would break if it were removed from the CPython sources?](https://github.com/python/cpython/blob/4844abdd700120120fc76c29d911bcb547700baf/Lib/os.py#L100)

Reply:
Well CPython doesn't do it in the native module, it just exports the `HAVE_x` constants that autoconf generates from the `posix`/`nt` module. But since the `os` functions don't necessarily directly correspond to native functions under the hood in RustPython, it's a bit easier to just generate it on the Rust side rather than making up fake constants.